benchmark/operations/create_container.py

```diff
@@ -0,0 +1,319 @@
+        service.
+    :param UUID node_uuid: Node on which to create dataset.
+    :param UUID dataset_id: ID for created dataset.
+    :param int volume_size: Size of volume in bytes.
+    :return Deferred[DatasetState]: The state of the created dataset.
+    """
+
+    def dataset_mounted(expected):
+        """
+        Check whether a dataset has been created and mounted.
+
+        :param Dataset expected: A dataset configuration to match against the
+            results of ``list_datasets_state``.
+        :return Deferred[Optional[DatasetState]]: The matching dataset state,
+            or None if the expected dataset is not found.
+        """
+        d = control_service.list_datasets_state()
+
+        def dataset_matches(inspecting, expected):
+            return (
+                expected.dataset_id == inspecting.dataset_id and
+                expected.primary == inspecting.primary and
+                inspecting.path is not None
+            )
+
+        def find_match(existing_state):
+            for state in existing_state:
+                if dataset_matches(state, expected):
+                    return state
+            return None
+        d.addCallback(find_match)
+        return d
+
+    d = control_service.create_dataset(
+        primary=node_uuid,
+        maximum_size=volume_size,
+        dataset_id=dataset_id,
+    )
+
+    def loop_until_dataset_mounted(expected):
+        return loop_until(reactor, partial(dataset_mounted, expected))
+    d.addCallback(loop_until_dataset_mounted)
+
+    return d
+
+
+def create_container(
+    reactor, control_service, node_uuid, name, image, volumes=None
+):
+    """
+    Create a container, then wait for it to be running.
+
+    :param IReactorTime reactor: Twisted Reactor.
+    :param IFlockerAPIV1Client control_service: Benchmark control
+        service.
+    :param UUID node_uuid: Node on which to start the container.
+    :param unicode name: Name of the container.
+    :param DockerImage image: Docker image for the container.
+    :param Optional[Sequence[MountedDataset]] volumes: Volumes to attach
+        to the container.
+    :return Deferred[ContainerState]: The state of the created container.
+    """
+
+    def container_started(expected):
+        """
+        Check whether a container has been created and started.
+
+        :param Container expected: A container configuration to match against
+            the results of ``list_containers_state``.
+        :return Deferred[Optional[ContainerState]]: The matching container
+            state, or None if the expected container is not found or is not
+            running.
+        """
+        d = control_service.list_containers_state()
+
+        def container_matches(inspecting, expected):
+            return (
+                expected.name == inspecting.name and
+                expected.node_uuid == inspecting.node_uuid and
+                inspecting.running is True
```

Comment:
Please just use `inspecting.running`. See https://www.python.org/dev/peps/pep-0008/#programming-recommendations (grep for "is True")